android/app/src/main/java/org/fossasia/openevent/viewmodels/auth/ChangePasswordActivityViewModel.java

```diff
@@ -0,0 +1,111 @@
+
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+import timber.log.Timber;
+
+public class ChangePasswordActivityViewModel extends ViewModel {
+
+    public static final int ON_ERROR = 2;
+    public static final int ON_COMPLETE = 3;
+
+    private RealmDataRepository realmRepo;
+    private User user;
+    private MutableLiveData<User> userLiveData;
+    private MutableLiveData<Integer> checkPasswordResponse;
+    private MutableLiveData<Integer> changePasswordResponse;
+    private Disposable checkPasswordDisposable;
+    private Disposable changePasswordDisposable;
+
+    public ChangePasswordActivityViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+    }
+
+    public LiveData<User> getUser() {
+        if (userLiveData == null) {
+            userLiveData = new MutableLiveData<>();
+            user = realmRepo.getUser();
+            user.addChangeListener(realmModel -> {
+                userLiveData.setValue(user);
+            });
+        }
+        return userLiveData;
+    }
+
+    public LiveData<Integer> checkCurrentPassword(String currentPassword) {
+        if (checkPasswordResponse == null) {
+            checkPasswordResponse = new MutableLiveData<>();
+        }
+        checkPasswordDisposable = APIClient.getOpenEventAPI().login(new Login(user.getEmail(), currentPassword))
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(loginResponse -> {
+                            Timber.d("Received token. Password is correct");
+                        },
+                        throwable -> {
+                            checkPasswordResponse.setValue(ON_ERROR);
+                            Timber.d("Error changing password: " + throwable.getMessage());
+                        },
+                        () -> {
+                            checkPasswordResponse.setValue(ON_COMPLETE);
+                        });
+        return checkPasswordResponse;
+    }
+
+    public LiveData<Integer> changePassword(String newPassword) {
+        if (changePasswordResponse == null) {
+            changePasswordResponse = new MutableLiveData<>();
+        }
+        int id = 0;
+        try {
+            id = JWTUtils.getIdentity(AuthUtil.getAuthorization());
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+        changePasswordDisposable = APIClient.getOpenEventAPI().updateUser(User.builder().id(id).password(newPassword).build(), id)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(user -> {
+                            RealmDataRepository
+                                    .getDefaultInstance()
+                                    .saveUser(user)
+                                    .subscribe();
```

Comment:
You can chain this call above using `mapCompletable` or equivalent above

Reply:
Or better yet, leave the responsibility of updating user on the `updateUser` method, moving the call there

Reply:
> Or better yet, leave the responsibility of updating user on the updateUser method, moving the call there

Sorry, i didn't understand what you mean by moving the call there. Can you explain a bit more.

Reply:
I have chained the calls using flatMapCompletable.